### PR TITLE
[SYCL-MLIR][Opaque] Fix `nd_item.get_group_linear_id` lowering

### PR DESCRIPTION
**Pull request by victor-eds**
Use correct base type for `llvm.getelementptr` operation: it should be `nd_item`, not `group`.

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]
Should we add a test from C++ to MLIR or LLVM IR?

**Comment by victor-eds** [comment]
> Should we add a test from C++ to MLIR or LLVM IR?

We have:
- [sycl-methods-to-llvm.mlir](https://github.com/intel/llvm/blob/8d4eff7c94a784781c81b5bb30a0fec3860f32ef/mlir-sycl/lib/Conversion/SYCLToLLVM/DPCPP.cpp) checking lowering to `llvm`
- [functions.cpp](https://github.com/intel/llvm/blob/8d4eff7c94a784781c81b5bb30a0fec3860f32ef/polygeist/tools/cgeist/Test/Verification/sycl/functions.cpp) checking MLIR codegen and that we can generate LLVM IR without breaking (involving the former lowering to the dialect and converting this to LLVM IR)

I think the union of these two is enough to cover this. WDYT?

File: mlir-sycl/lib/Conversion/SYCLToLLVM/DPCPP.cpp [hunk 1]
```diff
@@ -1978,10 +1978,12 @@ class NDItemGetGroupLinearIDPattern
         cast<NdItemType>(op.getNDItem().getType().getElementType())
             .getBody()[indices[0]];
     const auto convGroupTy = getTypeConverter()->convertType(groupTy);
+    Type ndItemTy = op.getNDItem().getType().getElementType();
```
Comment (whitneywhtsang) [inline mlir-sycl/lib/Conversion/SYCLToLLVM/DPCPP.cpp:1981]: [nit] Can you please change line 1977 and 1981 to:
```
NdItemType ndItemTy = cast<NdItemType>(op.getNDItem().getType().getElementType());
const auto groupTy = ndItemTy.getBody()[indices[0]];
```

Comment (victor-eds) [reply mlir-sycl/lib/Conversion/SYCLToLLVM/DPCPP.cpp]: Done

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**